main.tf

```diff
@@ -127,7 +145,7 @@ resource "aws_db_instance" "rds_master" {
   username                   = "${var.rds_instance_root_user_name}"
   password                   = "${var.rds_instance_root_user_password}"
   db_subnet_group_name       = "${var.rds_master_id == "" ? aws_db_subnet_group.rds_private_subnet.name : ""}"
-  parameter_group_name       = "${aws_db_parameter_group.rds_params.name}"
+  parameter_group_name       = "${var.max_connections != ""  ? aws_db_parameter_group.rds_params_with_max_connections.name : aws_db_parameter_group.rds_params_without_max_connections.name }"
```

Comment:
Another way I am wondering if we can consider is that any parent module that needs to set custom parameter group provisions the parameter group as required (in this case in the tf-product-listing repo) and sets the value of `parameter_group_name` with an override. So only databases that need it can explicitly set it in parent module after creating a custom parameter group.